src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1207,34 +1147,50 @@ private void ExecuteAndCatchIfThrows_NoLock<TArg>(DiagnosticAnalyzer analyzer, A
                 cancellationToken.ThrowIfCancellationRequested();
                 analyze(argument);
             }
-            catch (Exception e) when (ExceptionFilter(e, cancellationToken))
+            catch (Exception ex)
             {
-                // Diagnostic for analyzer exception.
-                var diagnostic = CreateAnalyzerExceptionDiagnostic(analyzer, e, info);
-                try
-                {
-                    _onAnalyzerException(e, analyzer, diagnostic, cancellationToken);
-                }
-                catch (Exception)
-                {
-                    // Ignore exceptions from exception handlers.
-                }
+                HandleAnalyzerException(ex, analyzer, info, OnAnalyzerException, _analyzerExceptionFilter, cancellationToken);
             }
         }
 
-        internal bool ExceptionFilter(Exception ex, CancellationToken cancellationToken)
+        internal static void HandleAnalyzerException(
```

Comment:
Contains all existing code now refactored to an internal static API so it can be invoked from the deprecated public API